plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/DefaultAccountProvider.java

```diff
@@ -25,30 +25,64 @@
 import com.google.api.client.http.javanet.NetHttpTransport;
 import com.google.api.client.json.JsonFactory;
 import com.google.api.services.oauth2.Oauth2;
+import com.google.auth.oauth2.GoogleAuthUtils;
 import com.google.cloud.tools.eclipse.googleapis.Account;
-import com.google.cloud.tools.eclipse.googleapis.IAccountProvider;
 import com.google.cloud.tools.eclipse.googleapis.UserInfo;
 import com.google.cloud.tools.eclipse.util.CloudToolsInfo;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import com.google.gson.stream.JsonReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
 import java.io.IOException;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Optional;
+import java.util.Timer;
+import java.util.TimerTask;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
 /**
  * 
  */
-public class DefaultAccountProvider implements IAccountProvider {
+public class DefaultAccountProvider extends AccountProvider {
 
+  private static final String ADC_PATH = GoogleAuthUtils.getWellKnownCredentialsPath();
+  public static final DefaultAccountProvider INSTANCE = new DefaultAccountProvider();
   private static final int USER_INFO_QUERY_HTTP_CONNECTION_TIMEOUT = 5000 /* ms */;
   private static final int USER_INFO_QUERY_HTTP_READ_TIMEOUT = 3000 /* ms */;
   private static final HttpTransport transport = new NetHttpTransport();
-  
+   
   private final JsonFactory jsonFactory = Utils.getDefaultJsonFactory();
   private final Logger LOGGER = Logger.getLogger(this.getClass().getName());
   
-  private Map<Credential, Account> accountCache = new HashMap<>();
+  private Optional<Credential> currentCred = computeCredential();
+  private Optional<Account> cachedAccount = Optional.empty();
+  private Timer adcPathPoller;
+  
+  private DefaultAccountProvider() {
+    adcPathPoller = new Timer();
+    TimerTask task = new TimerTask() {
+      @Override
+      public void run() {
+          checkIfAdcPathChanged();
+      }
+    };
+
+    // Schedule the task to run every second
+    adcPathPoller.scheduleAtFixedRate(task, 0, 1000);
```

Comment:
Is file-polling the recommended approach by the auth team?

Reply:
I reached out to auth to confirm. I noticed there is this [`WatchService`](https://docs.oracle.com/javase/8/docs/api/java/nio/file/WatchService.html) class as well - it is _intended to map directly on to the native file event notification facility where available_

Reply:
They got back saying that there is nothing built in in the library and a custom approach is fine as long as the credential is recreated.

Reply:
+1000 to using WatchService over the custom timer.

Reply:
Added Watch Service and tests